Src/Web/Web.Shared.Net/Extensibility/Implementation/PostSamplingTelemetryProcessor.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace Microsoft.ApplicationInsights.Web.Extensibility.Implementation
+{
+    using System.Web;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Common;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    /// <summary>
+    /// PostSamplingTelemetryProcessor evaluates deferred properties.
+    /// It is intended to be used with <see cref="RequestTrackingTelemetryModule.DisableTrackingProperties"/> after Sampling.
+    /// </summary>
+    /// <remarks>
+    /// This feature is still being evaluated and not recommended for end users.
+    /// This setting is not browsable at this time.
+    /// </remarks>
+    public class PostSamplingTelemetryProcessor : ITelemetryProcessor
+    {
+        private readonly ITelemetryProcessor nextProcessorInPipeline;
+        private TelemetryConfiguration telemetryConfiguration;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PostSamplingTelemetryProcessor"/> class.
+        /// </summary>
+        /// <param name="nextProcessorInPipeline">The next TelemetryProcessor in the chain.</param>
+        public PostSamplingTelemetryProcessor(ITelemetryProcessor nextProcessorInPipeline)
+        {
+            this.nextProcessorInPipeline = nextProcessorInPipeline;
+        }
+
+        private TelemetryConfiguration TelemetryConfiguration => this.telemetryConfiguration ?? (this.telemetryConfiguration = TelemetryConfiguration.Active);
+
+        /// <inheritdoc />
+        public void Process(ITelemetry item)
+        {
+            if (item is RequestTelemetry requestTelemetry)
+            {
+                var request = HttpContext.Current?.Request;
```

Comment:
nit: You can check that `if (requestTelemetry.Url == null)` before getting the current context. It will be a common mistake to include both processors

Reply:
It will not overwrite the Url & Source if already populated. We can add the check if Url or Source is null, then get request, but this should not be a concern, and I'd rather let pipeline handle that case. Let me know if you want this check